src/inmanta/ast/type.py

```diff
@@ -474,6 +619,35 @@ def __eq__(self, other: object) -> bool:
             return NotImplemented
         return self.element_type == other.element_type
 
+    def is_primitive(self) -> bool:
+        return self.get_base_type().is_primitive()
+
+    def corresponds_to(self, type: Type) -> bool:
+        if isinstance(type, Any):
+            return True
+
+        if not isinstance(type, List):
+            # Not a list at all
+            return False
+
+        if not isinstance(type, TypedList):
+            # The other list is untyped, so we are equivalent
+            return True
```

Comment:
Is this correct? When would this happen? Unparametrized `list` in Python would become `List(Any())` in the inmanta domain afaik.

Same question for dict.

Reply:
I agree, this is dead code. Type could/should be an ABC.

Same as above: do I raise an exception here?

Reply:
I'd honestly just return `False` if it's not a typed list, no?